File: core/src/main/java/de/qabel/core/repository/EntityManager.java
package de.qabel.core.repository;

import de.qabel.core.config.SyncSettingItem;

public class EntityManager extends GenericEntityManager<Integer, HasId> {
    public synchronized <T> void put(Class<T> entityType, SyncSettingItem entity) {
        put(entityType, entity, entity.getId());
    }

    @Override
    protected Integer getId(HasId entity) {
        return entity.getId();
    }
}

File: core/src/main/java/de/qabel/core/repository/EntityManager.kt
package de.qabel.core.repository

import de.qabel.core.config.SyncSettingItem

class EntityManager : GenericEntityManager<Int, HasId>() {
    @Synchronized fun <T> put(entityType: Class<T>, entity: SyncSettingItem) {
        put(entityType, entity, entity.id)
    }

    override fun getId(entity: HasId): Int? {
        return entity.id
    }
}
